server/game/cards/14.1-RaW/CycleOfRebirth.js

```diff
@@ -13,16 +13,44 @@ class CycleOfRebirth extends DrawCard {
                 cardCondition: card => card.type !== CardTypes.Province && card.type !== CardTypes.Stronghold
             },
             gameAction: AbilityDsl.actions.sequential([
-                AbilityDsl.actions.moveCard(context => ({
-                    destination: Locations.DynastyDeck,
-                    target: [context.target, context.source],
-                    shuffle: true
-                })),
+                AbilityDsl.actions.conditional({
+                    condition: context => context.target.controller === context.source.controller,
+                    trueGameAction: AbilityDsl.actions.moveCard(context => ({
+                        destination: Locations.DynastyDeck,
+                        target: [context.target, context.source],
+                        shuffle: true
+                    })),
+                    falseGameAction: AbilityDsl.actions.multiple([
+                        AbilityDsl.actions.moveCard(context => ({
+                            destination: Locations.DynastyDeck,
+                            target: context.target,
+                            shuffle: true
+                        })),
+                        AbilityDsl.actions.moveCard(context => ({
+                            destination: Locations.DynastyDeck,
+                            target: context.source,
+                            shuffle: true
+                        }))
+                    ])
+                }),
                 AbilityDsl.actions.refillFaceup(context => ({
                     target: [context.target.controller, context.source.controller],
                     location: [Locations.ProvinceOne, Locations.ProvinceTwo, Locations.ProvinceThree, Locations.ProvinceFour]
                 }))
-            ])
+            ]),
+            effect: 'shuffle {1}{3}{4} into {2}\'s dynasty deck{5}{6}{7}{8}{9}',
```

Comment:
Putting the ternary in effect instead of effectArgs would make this easier to follow (fewer args).

Reply:
effect doesn't take the context as an arg, so I don't know how to put the conditional in the effect itself.